### PR TITLE
Server WebSocket support

### PR DESCRIPTION
**Pull request by milgner**
Adds documentation to https://github.com/vert-x3/vertx-mqtt/pull/119

Note that the build process automatically aded `proxyProtocolTimeout` and `proxyProtocolTimeoutUnit` to the `dataobjects.adoc` file. I assume there is a reason for this and left them in...

### CODE REVIEW

**Comment by milgner** [comment]
I haven't changed anything in the code itself yet but this error makes me think I want to take another look:

>io.netty.handler.codec.DecoderException: too large message: 62 bytes
>	at io.netty.handler.codec.mqtt.MqttDecoder.decode(MqttDecoder.java:88)

**Comment by milgner** [comment]
The tests run fine locally, both with JRE 8 and 11. I saw that the exception visible on Travis CI is expected and part of a test. Not sure why it failed there. I'll try re-triggering the build and see what happens.

**Comment by milgner** [comment]
Seems like it was only a hiccup at Travis CI, everything passed now... :man_shrugging:

File: pom.xml [hunk 1]
```diff
@@ -143,6 +143,7 @@
           </includes>
           <excludes>
             <exclude>**/MqttServer*Test.java</exclude>
+            <exclude>**/MqttWebsocketServer*Test.java</exclude>
```
Comment (vietj) [inline pom.xml:146]: How about renaming the test to MqttServerWebSocketTest

**Comment by vietj** [comment]
there is a global issue with naming it should be spelled `WebSocket` and not `Websocket`, we had this inconsistency in Vert.x Core and we fixed it in master (we kept `WebsocketVersion` for backward compatiblity though).

**Comment by milgner** [comment]
Good point. I unified the naming, also with regards to the test class' names.

**Comment by milgner** [comment]
@vietj I'll see about resolving these new merge conflicts. Is there anything else I can do to help getting this merged?

**Comment by milgner** [comment]
The Travis failure seems to be present in the `master` branch, too, so I'd prefer to fix it separately from this branch.

**Comment by vietj** [comment]
I've merged and fixed issues with keep alive timeout, so now we should focus on getting this PR done and merged in master

**Comment by milgner** [comment]
I merged the current `master` into the feature branch - should be green now, hopefully 🤞

**Comment by milgner** [comment]
@vietj is there anything I can do to help getting this merged?

**Comment by vietj** [comment]
@milgner it seems there are conflicts "This branch cannot be rebased due to conflicts"

**Comment by vietj** [comment]
I scoped this branch to 4.0 milestone so it does not get forgotten

**Comment by milgner** [comment]
> @milgner it seems there are conflicts "This branch cannot be rebased due to conflicts"

Oh my. I had resolved conflicts through a merge recently but I can also rebase it for a fast-forward merge. 👍

**Comment by vietj** [comment]
@milgner can you provide a clean branch ? currently I cannot operate it on GitHub due to conflicts

**Comment by vietj** [comment]
Screen Shot 2020-12-02 at 09 49 01

**Comment by milgner** [comment]
@vietj the branch is rebased now and can be merged without conflicts again. Running CI took two attempts, though.

**Comment by vietj** [comment]
thanks this is green now and mergeable

**Comment by vietj** [comment]
sorry it was not merged for 4.0.0, it shall be for 4.0.1

**Comment by vietj** [comment]
@milgner could it be possible to have an integration/smoke test with a client different than Vert.x ? WDYT ? (we can do that in a separate PR after merge)

**Comment by vietj** [comment]
actually never mind, I can see it uses Paho

**Comment by vietj** [comment]
@milgner is it possible likewise to have WS support for client ?

**Comment by vietj** [comment]
thanks for the contribution @milgner !

**Comment by milgner** [comment]
> @milgner is it possible likewise to have WS support for client ?

I can take a look! So far we've been using Eclipse Paho as a client but maybe I can integrated WS support into the Vert.x client, too :+1:

**Pull request merged**